WeakAuras/Transmission.lua

```diff
@@ -730,283 +953,390 @@ local function scamCheck(codes, data)
+              for key in pairs(childDiff) do
+                local button = keyToButton[key]
+                if button and not info.tabsChanged[button] then
+                  info.tabsChanged[button] = true
+                  tabsChanged = tabsChanged + 1
                 end
+                if tabsChanged >= 7 then break end
               end
-
-              if (icons) then
-                tinsert(tooltip, {2, left, name..(icons[name] and (" |T"..icons[name]..":12:12:0:0:64:64:4:60:4:60|t") or ""), 1, 1, 1, 1, 1, 1});
-              else
-                local icon = WeakAuras.spellCache and WeakAuras.spellCache.GetIcon(name) or "Interface\\Icons\\INV_Misc_QuestionMark";
-                tinsert(tooltip, {2, left, name.." |T"..icon..":12:12:0:0:64:64:4:60:4:60|t", 1, 1, 1, 1, 1, 1});
-              end
-            end
-          elseif(trigger.type == "event" or trigger.type == "status") then
-            if(trigger.type == "event") then
-              tinsert(tooltip, {2, L["Trigger:"], (event_types[trigger.event] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
-            else
-              tinsert(tooltip, {2, L["Trigger:"], (status_types[trigger.event] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
-            end
-            if(trigger.event == "Combat Log" and trigger.subeventPrefix and trigger.subeventSuffix) then
-              tinsert(tooltip, {2, L["Message type:"], (WeakAuras.subevent_prefix_types[trigger.subeventPrefix] or L["Undefined"]).." "..(WeakAuras.subevent_suffix_types[trigger.subeventSuffix] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
             end
           else
-            tinsert(tooltip, {2, L["Trigger:"], L["Custom"], 1, 1, 1, 1, 1, 1});
+            info.unmodified = info.unmodified + 1
           end
         end
       end
+      return (modified or next(info.deleted)) and info or false
+    else
+      return modified and info or false
     end
+  end
+end
 
-    if (import and #tooltip > 30) then
-      -- Truncate the tooltip to ~25 auras if there are more than ~30
-      local size = #tooltip
-      tooltip[26] = {2, " ",  "[...]", 1, 1, 1, 1, 1, 1};
-      local nrOfChildren = children and #children or data.controlledChildren and #data.controlledChildren or 0
-      tooltip[27] = {1, string.format(L["%s total auras"], nrOfChildren), "", 1, 1, 1, 1};
-      for i = 28, size do
-        tooltip[i] = nil;
-      end
+local recurseUpdate
+function recurseUpdate(data, diff)
+  for k,v in pairs(diff) do
+    if v == deleted then
+      data[k] = nil
+    elseif type(v) == 'table' and type(data[k]) == 'table' then
+      recurseUpdate(data[k], v)
+    else
+      data[k] = v
     end
+  end
+end
 
+function WeakAuras.Update(data, diff)
+  -- modifies the installed data such that it matches the pending import, sans user-specified options
+  -- diff is expected to be output generated by WeakAuras.diff
+  if not diff or not data then 
+    return data 
+  end
+  WeakAuras.DeleteOption(data)
```

Comment:
Doesn't that mean that the auras are added at the end. E.g. if I have a group with children A B C, where I update B C, then the result is B C A ?